File: _assets/output_gen/Entities/Office.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class Office
    {
        //--------------------------------------------------------
        [Key]
        public int Id { get; set;}

        [Required, StringLength(64)]
        public string Name { get; set;}

        [StringLength(2000)]
        public string Description { get; set;}

        public DateTime DateCreated { get; set;}

        public DateTime DateUpdated { get; set;}

        [StringLength(32)]
        public string UpdatedBy { get; set;}

        public List<Candidate> Candidates { get; set;}

        //--------------------------------------------------------
    }
}

File: _assets/output_gen/jsModels/Office.js

//---------------------------------------------------------------------------

export default class Office {
  constructor(options) {

    this.Id = null;
    this.Name = null;
    this.Description = null;
    this.DateCreated = null;
    this.DateUpdated = null;
    this.UpdatedBy = null;
    this.Candidates = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.Name = options.Name || this.Name;
    this.Description = options.Description || this.Description;
    this.DateCreated = options.DateCreated || this.DateCreated;
    this.DateUpdated = options.DateUpdated || this.DateUpdated;
    this.UpdatedBy = options.UpdatedBy || this.UpdatedBy;
    this.Candidates = options.Candidates || this.Candidates;

  }  
  //---------------------------------------------------------------------------  
}